schematools/maps/generators/mapfile.py

```diff
@@ -0,0 +1,133 @@
+from collections import ChainMap
+from dataclasses import dataclass
+import typing
+from amsterdam_schema.types import DatasetSchema, DatasetTableSchema
+
+# XXX bring in line with new amsterdam_schema ??
+# amsterdam_schema was changed backwards incompatible
+# now just moved the old schema def inside this package
+# from dataservices import amsterdam_schema as schema
+from ..interfaces.mapfile import types, serializers
+
+
+MapfileStr = str
+
+
+@dataclass
+class MapserviceLayerContext:
+    """ Context for generating mapfile layers from a amsterdam datatable
+        and its parent dataset. """
+
+    dataset: DatasetSchema
+    datatable: DatasetTableSchema
+
+    @property
+    def name(self):
+        return self.datatable.id
+
+    @property
+    def projections(self) -> typing.List[str]:
+        return [self.dataset["crs"]]
+
+    @property
+    def srid(self) -> int:
+        return int(self.dataset["crs"].split(":")[-1])  # XXX replace with our impl.
+
+    @property
+    def metadata(self) -> dict:
+        return {"ows_enable_request": "*"}
+
+
+class MapServiceContext:
+    """ Context for generating mapfiles from a amsterdam dataset """
+
+    def __init__(self, dataset: DatasetSchema):
+        self.dataset = dataset
+
+    @property
+    def name(self):
+        return self.dataset.id
+
+    @property
+    def layers(self) -> typing.List[MapserviceLayerContext]:
+        return [MapserviceLayerContext(self.dataset, dt) for dt in self.dataset.tables]
+
+
+@dataclass
+class Generator:
+    def serialize(self, dataset) -> str:
+        raise NotImplementedError
+
+    def __call__(self, dataset):
+        return self.serialize(dataset)
```

Comment:
Is misschien overbodig? Zonder `__call__` toepassing zie je duidelijker waar en hoe de generator aangeroepen wordt.